pkg/cloudprovider/cloud.go

```diff
@@ -123,14 +123,22 @@ type Instances interface {
 	CurrentNodeName(hostname string) (string, error)
 }
 
+type Instance struct {
+	// Name is the name of the instance as Kubernetes understands it. Usually the hostname
+	Name string
+
+	// ID is the instance ID. To be used for clouds that don't have a mapping of hostname to instance
+	ID string
```

Comment:
where are you using this? or are you maintaining a cloudprovider elsewhere that you compile into the master